### PR TITLE
Port builder_basics from gtk-rs/examples4

### PR DESCRIPTION
**Pull request by vamsikrishna-brahmajosyula**

### CODE REVIEW

File: examples/src/bin/builder_basics.rs [hunk 1]
```diff
@@ -0,0 +1,49 @@
+//! # Builder Basics Sample
+//!
+//! This sample demonstrates how to use the builder with an imported glade file
+
+extern crate gio;
+extern crate gtk;
+
```
Comment (bilelmoussaoui) [inline examples/src/bin/builder_basics.rs:7]:
```suggestion
```

File: examples/src/bin/builder_basics.rs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    let window: ApplicationWindow = builder.get_object("window1").expect("Couldn't get window1");
+    window.set_application(Some(application));
+    let bigbutton: Button = builder.get_object("button1").expect("Couldn't get button1");
+    let dialog: MessageDialog = builder
+        .get_object("messagedialog1")
+        .expect("Couldn't get messagedialog1");
+
+    dialog.connect_response(move |d: &MessageDialog, _: ResponseType| {
+        d.hide();
```
Comment (bilelmoussaoui) [inline examples/src/bin/builder_basics.rs:27]:
```suggestion
        d.close();
```

File: examples/src/bin/builder_basics.glade [hunk 3]
```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (bilelmoussaoui) [inline examples/src/bin/builder_basics.glade:1]: Use .ui for this file instead of .glade

File: examples/src/bin/builder_basics.glade [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        <property name="tooltip_markup" translatable="yes">Thank you for trying this example</property>
+        <property name="orientation">vertical</property>
+        <property name="spacing">2</property>
+        <child internal-child="action_area">
+          <object class="GtkBox" id="messagedialog-action_area1">
+            <child>
+              <placeholder/>
+            </child>
+          </object>
+        </child>
```
Comment (bilelmoussaoui) [inline examples/src/bin/builder_basics.glade:32]:
```suggestion
```

File: examples/src/bin/builder_basics.ui [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+      <object class="GtkButton" id="button1">
+        <property name="label" translatable="yes">Big Useless Button</property>
+      </object>
+    </child>
+  </object>
+  <object class="GtkMessageDialog" id="messagedialog1">
+    <property name="transient-for">window1</property>
+    <property name="resizable">0</property>
+    <property name="modal">1</property>
+    <property name="hide-on-close">1</property>
```
Comment (bilelmoussaoui) [inline examples/src/bin/builder_basics.ui:17]: Those are booleans, use True/False instead please

File: examples/src/bin/builder_basics.ui [hunk 6]
```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<interface>
+  <object class="GtkApplicationWindow" id="window1">
```
Comment (bilelmoussaoui) [inline examples/src/bin/builder_basics.ui:3]:
```suggestion
  <object class="GtkApplicationWindow" id="window">
```

File: examples/src/bin/builder_basics.ui [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    <property name="title" translatable="yes">Builder Basics</property>
+    <property name="default_width">320</property>
+    <property name="default_height">240</property>
+    <child>
+      <object class="GtkButton" id="button1">
```
Comment (bilelmoussaoui) [inline examples/src/bin/builder_basics.ui:8]:
```suggestion
      <object class="GtkButton" id="button">
```

File: examples/src/bin/builder_basics.ui [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+        <property name="label" translatable="yes">Big Useless Button</property>
+      </object>
+    </child>
+  </object>
+  <object class="GtkMessageDialog" id="messagedialog1">
```
Comment (bilelmoussaoui) [inline examples/src/bin/builder_basics.ui:13]:
```suggestion
  <object class="GtkMessageDialog" id="message_dialog">
```

File: examples/src/bin/builder_basics.ui [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+    <property name="transient-for">window1</property>
```
Comment (bilelmoussaoui) [inline examples/src/bin/builder_basics.ui:14]:
```suggestion
    <property name="transient-for">window</property>
```

File: examples/src/bin/builder_basics.ui [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+    <property name="buttons">ok</property>
+    <child internal-child="message_area">
+      <object class="GtkBox" id="messagedialog-vbox1">
+        <property name="name">msgdialog</property>
+        <property name="width_request">300</property>
+        <property name="tooltip_markup" translatable="yes">Thank you for trying this example</property>
+        <property name="orientation">vertical</property>
+        <property name="spacing">2</property>
+        <child>
+          <object class="GtkLabel" id="label2">
```
Comment (bilelmoussaoui) [inline examples/src/bin/builder_basics.ui:27]:
```suggestion
          <object class="GtkLabel">
```

File: examples/src/bin/builder_basics.ui [hunk 11]
```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (bilelmoussaoui) [inline examples/src/bin/builder_basics.ui:1]: I would put the ui file in a ui subdirectory though as we will probably have more of them in the future for other examples

File: examples/src/bin/builder_basics.rs [hunk 12]
```diff
@@ -0,0 +1,46 @@
+//! # Builder Basics Sample
+//!
+//! This sample demonstrates how to use the builder with an imported glade file
+
+use gio::prelude::*;
+use gtk::prelude::*;
+
+use gtk::{ApplicationWindow, Builder, Button, MessageDialog, ResponseType};
```
Comment (bilelmoussaoui) [inline examples/src/bin/builder_basics.rs:8]: Any reason for not doing the same for Application as well ?

File: examples/src/bin/builder_basics.rs [hunk 13]
```diff
@@ -0,0 +1,46 @@
+//! # Builder Basics Sample
+//!
+//! This sample demonstrates how to use the builder with an imported glade file
```
Comment (bilelmoussaoui) [inline examples/src/bin/builder_basics.rs:3]:
```suggestion
//! This sample demonstrates how to use the builder with an imported UI file
```
glade is history

File: examples/src/bin/ui/builder_basics.ui [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+        <child>
+          <object class="GtkLabel">
+            <property name="label" translatable="yes">You have pressed the button</property>
+            <property name="ellipsize">end</property>
+            <property name="width_chars">40</property>
+            <property name="lines">1</property>
+          </object>
+        </child>
+      </object>
+    </child>
```
Comment (bilelmoussaoui) [inline examples/src/bin/ui/builder_basics.ui:35]:
```suggestion
    <property name="text" translatable="yes">Thank you for trying this example</property>
    <property name="secondary-text" translatable="yes">You have pressed the button</property>
```
That's just glade over-doing things

File: examples/src/bin/ui/builder_basics.ui [hunk 15]
```diff
@@ -0,0 +1,37 @@
...
+      <object class="GtkButton" id="button">
+        <property name="label" translatable="yes">Big Useless Button</property>
+      </object>
+    </child>
+  </object>
+  <object class="GtkMessageDialog" id="messagedialog">
+    <property name="transient-for">window</property>
+    <property name="resizable">False</property>
+    <property name="modal">True</property>
+    <property name="hide-on-close">True</property>
```
Comment (bilelmoussaoui) [inline examples/src/bin/ui/builder_basics.ui:17]:
```suggestion
```

File: examples/src/bin/builder_basics.rs [hunk 16]
```diff
@@ -0,0 +1,46 @@
...
+use gio::prelude::*;
+use gtk::prelude::*;
+
+use gtk::{Application, ApplicationWindow, Builder, Button, MessageDialog, ResponseType};
+
+use std::env::args;
+
+fn build_ui(application: &Application) {
+    let glade_src = include_str!("ui/builder_basics.ui");
+    let builder = Builder::from_string(glade_src);
```
Comment (bilelmoussaoui) [inline examples/src/bin/builder_basics.rs:14]:
```suggestion
    let ui_src = include_str!("ui/builder_basics.ui");
    let builder = Builder::from_string(ui_src);
```
We shouldn't mention glade at all, it's not gtk4 compatible

**Comment by bilelmoussaoui** [comment]
Other than those latest comments, lgtm. Thanks!

**Comment by bilelmoussaoui** [comment]
Looks good to me! thanks

**Comment by bilelmoussaoui** [comment]
Feel free to port other examples in the future ;)

**Comment by vamsikrishna-brahmajosyula** [comment]
Sure, will port as many as I can.

**Pull request merged**